express_relay/sdk/js/src/index.ts

```diff
@@ -591,4 +610,142 @@ export class Client {
       return response.data;
     }
   }
+
+  /**
+   * Adds a SubmitBid instruction to a transaction, thereby permissioning it
+   * @param txRaw The transaction to add a SubmitBid instruction to. This transaction should already check for the appropriate permissions.
+   * @param protocol The identifying address of the protocol that the permission key is for
+   * @param permissionKey The 32-byte permission key as an SVM PublicKey
+   * @param bidAmount The amount of the bid in lamports
+   * @param deadline The deadline for the bid in seconds since Unix epoch
+   * @param secretKey The secret key of the searcher that will serve as fee payer for the transaction
+   * @param chainId The chain ID as a string, e.g. "solana"
+   * @returns The new transaction with the SubmitBid instruction
+   */
+  async constructPermissionedTxFromSvmTx(
+    txRaw: Transaction,
+    protocol: PublicKey,
+    permissionKey: PublicKey,
+    bidAmount: anchor.BN,
+    deadline: anchor.BN,
+    secretKey: Uint8Array,
+    chainId: string
+  ): Promise<Transaction> {
+    const searcher = Keypair.fromSecretKey(secretKey);
+
+    if (this.clientOptions.svmEndpoints === undefined) {
+      throw new Error("SVM endpoints not provided");
+    }
+    const connectionSvm = new Connection(
+      new URL(this.clientOptions.svmEndpoints[chainId]).toString(),
+      "confirmed"
+    );
+    const provider = new AnchorProvider(
+      connectionSvm,
+      new anchor.Wallet(searcher),
+      {}
+    );
+    const expressRelay = new Program<ExpressRelay>(
+      expressRelayIdl as ExpressRelay,
+      provider
+    );
+
+    const svmConstants = SVM_CONSTANTS[chainId];
+
+    const protocolConfig = PublicKey.findProgramAddressSync(
+      [anchor.utils.bytes.utf8.encode("config_protocol"), protocol.toBuffer()],
+      svmConstants.expressRelayProgram
+    )[0];
+
+    let feeReceiverProtocol: PublicKey;
+    const protocolAccountInfo = await connectionSvm.getAccountInfo(protocol);
+    let protocolExecutable = false;
+    if (protocolAccountInfo) {
+      protocolExecutable = protocolAccountInfo.executable;
+    }
+    if (protocolExecutable) {
+      feeReceiverProtocol = PublicKey.findProgramAddressSync(
+        [anchor.utils.bytes.utf8.encode("express_relay_fees")],
+        protocol
+      )[0];
+    } else {
+      feeReceiverProtocol = protocol;
+    }
+
+    const expressRelayMetadata = PublicKey.findProgramAddressSync(
+      [anchor.utils.bytes.utf8.encode("metadata")],
+      svmConstants.expressRelayProgram
+    )[0];
+
+    const ixSubmitBid = await expressRelay.methods
+      .submitBid({
+        deadline,
+        bidAmount,
+      })
+      .accountsPartial({
```

Comment:
you can use `accounts` instead of `accountsPartial` and remove anything that's not predetermined which makes the logic simpler

Reply:
i don't want to upload a new IDL for each new program ID (e.g. if we have deployments on a different SVM chain). Instead i'd rather include the new program ID for the new chain in SVM_CONSTANTS and manually construct all the accounts in the client